### ISSUE TITLE
Ordered lists copied to clipboard as unordered lists

### ISSUE THREAD

**Issue by ashucg** [marktext/marktext#2298]
### Description
Creating an ordered list doesn't use correct marker unless you export it, click on empty area or press return key multiple time to exit ordered list. I have created a screencast demonstrating it.

### Steps to reproduce

1. Start document by creating an ordered list
2. Add some points
3. Select all text and copy the text or markdown - document should only contain ordered list, don't click on the empty area or press return key multiple times to exit ordered list
4. Now paste it in a new editor or anywhere you can see the copied content

**Expected behavior:**

Should respect settings and use correct marker for ordered list.

**Actual behavior:**

Wrong marker is used for ordered list.

**Link to an example:**

[Check this screencast](https://streamable.com/n2sj0c)

**Current Marker Settings:**
![image](https://user-images.githubusercontent.com/11291284/89554273-466e0f00-d82c-11ea-86ed-ea85e3d4c10e.png)

### Versions

- Mark Text: v0.16.2
- Operating system: Fedora 32

**Comment by brainchild0** [comment]
Is the issue one that we essentially  might describe as "Ordered lists copied to clipboard as unordered lists"?

**Comment by ashucg** [comment]
@brainchild0 Yeah, that's correct. I just tried switching to **Source Code Mode** and I can see the correct marker there so it is only happening when content is being copied to clipboard.

**Comment by Jocs** [comment]
Thank you for the detailed report and screencast.

This bug was caused by the old editor engine (muyajs) reading list-item marker metadata directly from live DOM nodes at copy time — when the cursor was still inside the list, the ordered-list marker hadn't been flushed to the DOM, so the serializer fell through to a `-` default.

MarkText has since been migrated to a new editor engine (`@muyajs/core`). The new clipboard path (`src/clipboard/copyData.ts`) never reads from the DOM selection; it serializes directly from the typed block-tree state. `OrderList.getState()` always returns the correct `{ meta: { delimiter, start, loose } }` regardless of cursor position, and `StateToMarkdown._serializeListItem()` discriminates ordered from bullet lists via the typed state and emits `N. ` / `N) ` markers correctly. This is covered by the `listMarkerOption.spec.ts` test suite.

Closing as resolved by the engine migration.
